parfive/downloader.py

```diff
@@ -424,20 +425,21 @@ async def _get_http(self, session, *, url, filepath_partial, chunksize=100,
                         for _range in ranges:
                             download_workers.append(
                                 self.loop.create_task(self._http_download_worker(
-                                    session, url, chunksize, _range, timeout, downloaded_chunk_queue, **kwargs
+                                    session, url, chunksize, _range, timeout, downloaded_chunks_queue, **kwargs
                                 ))
                             )
                     else:
                         download_workers.append(
                             self.loop.create_task(self._http_download_worker(
-                                session, url, chunksize, None, timeout, downloaded_chunk_queue, **kwargs
+                                session, url, chunksize, None, timeout, downloaded_chunks_queue, **kwargs
                             ))
                         )
 
+                    # join() waits till all the items in the queue have been processed
+                    await downloaded_chunks_queue.join()
                     # run all the download workers
                     await asyncio.gather(*download_workers)
-                    # join() waits till all the items in the queue have been processed
-                    await downloaded_chunk_queue.join()
```

Comment:
I swapped them as test but it doesn't seem to break anything.

I saw this: https://docs.python.org/3/library/asyncio-queue.html and they do the same but at this point im clutching at straws.